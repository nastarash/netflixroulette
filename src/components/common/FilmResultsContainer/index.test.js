import React from 'react';
import { FilmResultsContainer, FilmResults } from './index';

describe('<FilmReslutsContainer />', () => {
  test('rendered correctly in normal state', () => {
    const filmResultsContainer = shallow(
        <FilmResultsContainer />
    );
    expect(filmResultsContainer).toMatchSnapshot();
  });


  test('<FilmResults /> rendered correctly in error state', () => {
    const fetchDataMockFn = jest.fn();
    const films = [];

    const filmResultsContainerWithError = shallow(
        <FilmResults hasErrored={true} fetchData={fetchDataMockFn} films={films} />
    );
    expect(filmResultsContainerWithError).toMatchSnapshot();
  });


  test('<FilmResults /> rendered correctly in normal state', () => {
    const fetchDataMockFn = jest.fn();
    const films = {
      records: [{ id: 11,
        title: 'Star Wars',
        tagline: 'A long time ago in a galaxy far, far away...',
        vote_average: 8.1,
        vote_count: 8516,
        release_date: '1977-05-25',
        poster_path: 'https://image.tmdb.org/t/p/w500/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg',
        overview: 'Princess Leia is captured and held hostage by ' +
                  'the evil Imperial forces in their effort to take ' +
                  'over the galactic Empire.',
        budget: 11000000,
        revenue: 775398007,
        genres: ['Adventure', 'Action', 'Science Fiction'],
        runtime: 121,
      }],
    };

    const filmResultsContainerWithError = shallow(
        <FilmResults hasErrored={false} fetchData={fetchDataMockFn} films={films} />
    );
    expect(filmResultsContainerWithError).toMatchSnapshot();
  });


  test('<FilmResults /> rendered correctly in loading state', () => {
    const fetchDataMockFn = jest.fn();
    const films = { records: []};

    const filmResultsContainerWithError = shallow(
        <FilmResults isLoading={true} fetchData={fetchDataMockFn} films={films} />
    );
    expect(filmResultsContainerWithError).toMatchSnapshot();
  });


  test('calls fetchData if films.length === 0', () => {
    const fetchDataMockFn = jest.fn();
    const films = [];

    shallow(
        <FilmResults isLoading={true} fetchData={fetchDataMockFn} films={films} />
    );

    expect(fetchDataMockFn).toBeCalled();
  });
});
