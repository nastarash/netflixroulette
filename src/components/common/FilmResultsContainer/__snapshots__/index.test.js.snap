// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmReslutsContainer /> <FilmResults /> rendered correctly in error state 1`] = `
<h1>
  Ooops, there is an error
</h1>
`;

exports[`<FilmReslutsContainer /> <FilmResults /> rendered correctly in loading state 1`] = `
<span>
  loading...
</span>
`;

exports[`<FilmReslutsContainer /> <FilmResults /> rendered correctly in normal state 1`] = `
<div
  className="film-results-container"
>
  <withRouter(FilmResultBody)
    filmResult={
      Object {
        "budget": 11000000,
        "genres": Array [
          "Adventure",
          "Action",
          "Science Fiction",
        ],
        "id": 11,
        "overview": "Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire.",
        "poster_path": "https://image.tmdb.org/t/p/w500/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg",
        "release_date": "1977-05-25",
        "revenue": 775398007,
        "runtime": 121,
        "tagline": "A long time ago in a galaxy far, far away...",
        "title": "Star Wars",
        "vote_average": 8.1,
        "vote_count": 8516,
      }
    }
    key="11"
    onClick={[Function]}
  />
</div>
`;

exports[`<FilmReslutsContainer /> rendered correctly in normal state 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
